File: lib/layer2_core/dark/e_on_surface_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSurfaceDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd4;
      case 'standard': return cstd4;
      case 'blue': return cbd4;
      case 'green': return cgd4;
      case 'red': return crd4;
      case 'nothing': return cnod4;
      default: return cstd4;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSurfaceDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSurfaceDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd4
            AppTheme.STANDARD -> R.color.cstd4
            AppTheme.BLUE -> R.color.cbd4
            AppTheme.GREEN -> R.color.cgd4
            AppTheme.RED -> R.color.crd4
            AppTheme.NOTHING -> R.color.cnod4
        }
    }
}
